packages/teleport/src/AppLauncher/AppLauncher.tsx

```diff
@@ -0,0 +1,37 @@
+/*
+Copyright 2019 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React from 'react';
+import { Flex, Indicator } from 'design';
+import { AccessDenied } from 'design/CardError';
+import useAppLauncher from './useAppLauncher';
+
+export default function Container() {
+  const state = useAppLauncher();
+  return <AppLauncher {...state} />;
+}
+
+export function AppLauncher(props: ReturnType<typeof useAppLauncher>) {
+  if (props.status === 'failed') {
+    return <AccessDenied message={props.statusText} />;
+  }
+
+  return (
+    <Flex height="180px" justifyContent="center" alignItems="center" flex="1">
+      <Indicator />
```

Comment:
i think we should put a message saying "redirecting"? i ran these in storybook and the blank states was not very helpful. maybe there is a polish coming up.